File: ms-probe/src/main/kotlin/br/com/lteixeira/msprobe/application/gateway/message/model/Probe.kt
package br.com.lteixeira.msprobe.application.gateway.message.model

import br.com.lteixeira.msprobe.application.enumeration.Status

data class Probe(
    val id: String,
    val planet: String,
    val probe: String,
    val probeCoordinate: ProbeCoordinate,
    val status: Status
)

data class ProbeCoordinate(
    val locationX: Int,
    val locationY: Int,
)

File: ms-radar/internal/model/probe.go
package model

type ProbeCoordinate struct {
	LocationX int64 `json:"locationX" validate:"required"`
	LocationY int64 `json:"locationY" validate:"required"`
}

type Probe struct {
	Name            string          `json:"probe" validate:"required"`
	Planet          string          `json:"planet" validate:"required"`
	ProbeCoordinate ProbeCoordinate `json:"probeCoordinate" validate:"required"`
	Status          string          `json:"status" validate:"required"`
}
